src/mon/MonMap.h

```diff
@@ -56,15 +56,17 @@ struct mon_info_t {
    * the priority of the mon, the lower value the more preferred
    */
   uint16_t priority{0};
+  uint16_t weight{0};
```

Comment:
i'd suggest use a non-zero weight. so the default behavior is to uniform distribution. and that's expected. also, no need to make this a special case.